### ISSUE TITLE
Lint Rule: Invalid topic name for `Type: AWS::SNS::Topic`

### ISSUE THREAD

**Issue by DeflateAwning** [aws-cloudformation/cfn-lint#3826]
### Is this feature request related to a new rule or cfn-lint capabilities?

rules

### Describe the feature you'd like to request

This resource should fail:

```yaml

Resources:
  EmailNotificationActionTopic:
    Type: AWS::SNS::Topic
    Properties:
      Subscription:
        - Endpoint: "me@example.com"
          Protocol: email
      TopicName: Company A's Cool Notifications
```

### Describe the solution you'd like

Example above should fail

### Is this something that you'd be interested in working on?

- [ ] 👋 I may be able to implement this feature request

### Would this feature include a breaking change?

- [ ] ⚠️ This feature might incur a breaking change
